tests/Backend/Snowflake/DirectAccessTest.php

```diff
@@ -227,6 +228,38 @@ public function testDirectAccessAndRestrictions()
         }));
         $this->assertSame('mytable', $views[0]['name']);
 
+        try {
+            $bucketUpdateOptions = new BucketUpdateOptions($bucketId, 'updated-b1-display-name', false);
+            $this->_client->updateBucket($bucketUpdateOptions);
+            $this->fail('Should have thrown!');
+        } catch (ClientException $e) {
+            $this->assertSame(
+                'Bucket "in.c-API-tests" has Direct Access enabled please use async call',
+                $e->getMessage()
+            );
+        }
+
+        $bucketUpdateOptions = new BucketUpdateOptions($bucketId, 'updated-b1-display-name', true);
+        $this->_client->updateBucket($bucketUpdateOptions);
+
+        $schemas = $connection->fetchAll('SHOW SCHEMAS');
+        $this->assertCount(2, $schemas, 'There should be INFORMATION SCHEMA and one bucket');
+        $schemas = array_values(array_filter($schemas, function ($schema) {
+            return $schema['name'] === 'DA_IN_UPDATED-B1-DISPLAY-NAME';
+        }));
+        $this->assertSame('DA_IN_UPDATED-B1-DISPLAY-NAME', $schemas[0]['name']);
+
+        $connection->query(sprintf(
+            'USE SCHEMA %s',
+            $connection->quoteIdentifier($schemas[0]['name'])
+        ));
+
+        $viewsResult = $connection->fetchAll('SHOW VIEWS');
+        $this->assertCount(1, $viewsResult);
+        $views = array_values(array_filter($viewsResult, function ($view) {
+            return $view['name'] === 'mytable';
+        }));
+        $this->assertSame('mytable', $views[0]['name']);
```

Comment:
pro jistotu bych tady přihodil query na select * from mytable - jestli se tím renamem nezměnily ty future grants - tj. jestli jsou pověšené na nějaké interní ID schematu, nebo jestli jsou na názvu... (a já mám obavu, že jsou na názvu)

Reply:
Podla mna to funguje spravne ten update, podla mna bude lepsie ked robime update ako by sme to cele zapli a vypli. Pridal som tam ten selekt co pises a dostanem tu tabulku ktora je naplnena